core/src/main/java/org/infinispan/xsite/irac/DefaultIracManager.java

```diff
@@ -201,7 +218,7 @@ public void onTopologyUpdate(CacheTopology oldCacheTopology, CacheTopology newCa
       if (!addedSegments.isEmpty()) {
          // request state from old primary owner
          // and group new segments by primary owner
-         Map<Address, IntSet> primarySegments = new HashMap<>();
+         Map<Address, IntSet> primarySegments = new HashMap<>(newCacheTopology.getMembers().size());
          for (int segment : addedSegments) {
             Address primary = newCacheTopology.getWriteConsistentHash().locatePrimaryOwnerForSegment(segment);
             primarySegments.computeIfAbsent(primary, DefaultIracManager::newIntSet).add(segment);
```

Comment:
```suggestion
            primarySegments.computeIfAbsent(primary, DefaultIracManager::newIntSet).set(segment);
```
To be honest this is an issue in a bunch of places, that we should fix as well. I logged ISPN-13604 where I will include this if you don't change it.

Reply:
https://github.com/infinispan/infinispan/pull/9772 fixes this btw.